velox/connectors/Connector.h

```diff
@@ -287,6 +287,67 @@ class DataSource {
   }
 };
 
+/// Represents a lookup request for a given input.
+struct LookupRequest {
+  /// Contains the input column vectors used by lookup join and range
+  /// conditions.
+  RowVectorPtr input{nullptr};
+
+  LookupRequest(RowVectorPtr input) : input(std::move(input)) {}
+};
+
+/// Represents the lookup result for a subset of input produced by the
+/// 'LookupResultIterator'.
+struct LookupResult {
+  /// Used to specify the match result for a subset of input rows. It provides
+  /// an array of indices of the matched output rows in 'lookupOutput'. Each
+  /// array element contains all the matched rows for a given input row. If it
+  /// is empty, then there is no match for the given input row. 'inputHits'
+  /// contains the input indices in the same order of the corresponding input
+  /// rows in the lookup request. 'LookupResultIterator' also produces the
+  /// lookup result in the input request row order, and produces all the output
+  /// matches for a given input row in one lookup result.
+  ArrayVectorPtr inputHits;
+
+  /// The output vector that contains the lookup result rows.
+  RowVectorPtr lookupOutput{nullptr};
+
+  LookupResult(ArrayVectorPtr _inputHits, RowVectorPtr _lookupOutput)
+      : inputHits(std::move(_inputHits)),
+        lookupOutput(std::move(_lookupOutput)) {
+    VELOX_CHECK_EQ(
+        inputHits->elements()->typeKind(),
+        TypeKind::INTEGER,
+        "Uexpected lookup input hit index type");
+  }
+};
+
+/// The lookup result iterator used to fetch the lookup result in batch for a
+/// given lookup request.
+class LookupResultIterator {
+ public:
+  virtual ~LookupResultIterator() = default;
+
+  /// Returns nullptr if all the lookup results have been fetched. Returns
+  /// std::nullopt and sets the 'future' if started asynchronous work and needs
+  /// to wait for it to complete to continue processing. The caller will wait
+  /// for the 'future' to complete before calling 'next' again.
+  virtual std::optional<LookupResult> next(
+      uint64_t size,
+      velox::ContinueFuture& future) = 0;
+};
+
+class IndexSource {
+ public:
+  virtual ~IndexSource() = default;
+
+  /// Lookup the index source for the given request and returns an iterator to
```

Comment:
> Lookup the index source for the given request

This repeats the signature without providing any clarifying details. Is there something more useful to put in the comment?